src/app/js/formats/distributionChart/global-radar-chart/RadarChartAdmin.js

```diff
@@ -75,10 +76,86 @@ class RadarChartAdmin extends Component {
         updateAdminArgs('scale', scale, this.props);
     };
 
+    setColors = (_, colors) => {
+        updateAdminArgs('colors', colors, this.props);
+    };
+
+    setColorsWithColorPicker = () => {
+        updateAdminArgs('colors', this.getColorsFromPicker(), this.props);
+    };
+
+    getColorsFromPicker() {
+        let numberOfPickers = this.getColorsArray().length;
```

Comment:
Yeah the length was an issue !
I fixed it by adjusting the array size during my UI updating to match the correct size : 
For now I got this better version, because of the length of `state.colors` problem :
```javascript
            [...this.state.colors] = this.state.colors.slice(
                0,
                colorsArray.length,
            );
```